### PR TITLE
Support line breaks at the end of anchors, issue 99

### LINKED ISSUE #99: Linebreak after link gets removed
If there is a link before linebreak, the linebreak gets removed from markdown.

Current behavior (https://docs.google.com/document/d/1Au7MXm2-TRoS4M0UFBF-7Asiia-NCRZpce9eYV9hLlw/edit):
image

Corresponding HTML:
image

Expected behavior:
image

I could take a shot at this one as well.

### PR DESCRIPTION
**Pull request by trustedtomato**
Resolves #99 
Let me know if I should change something.

### CODE REVIEW

File: lib/fix-google-html.js [hunk 1]
```diff
@@ -255,6 +255,28 @@ export function unwrapLineBreaks (node) {
...
 
+/**
+ * Moves linebreaks outside of anchor elements,
+ * if the linebreak is the first and/or last child of the anchor.
+ * @param {RehypeNode} node 
+ */
+export function moveLinebreaksOutsideOfAnchors (node) {
+  visit(node, 'element', (node, index, parent) => {
+    console.log(node)
+    if (node.tagName === 'a') {
```
Comment (Mr0grog) [inline lib/fix-google-html.js:266]: You can limit to `<a>` elements by using:

```js
visit(node, n => n.tagName === 'a', (node, index, parent) => {
  // Code to deal with `<a>` elements here.
});
```

Or, if useful, some of these checks are defined and given a name at the top of the file: https://github.com/Mr0grog/google-docs-to-markdown/blob/a2690fe56b41d17dac16a8a686a29ac0975c2179/lib/fix-google-html.js#L67-L71

So you can do:

```js
// At top of file:
const isAnchor = (node) -> node.tagName === 'a';

// In your function:
visit(node, isAnchor, (node, index, parent) => {
  // Code to deal with `<a>` elements here.
});
```

---

Also please clean up the `console.log` calls!

File: lib/fix-google-html.js [hunk 2]
```diff
@@ -255,6 +255,28 @@ export function unwrapLineBreaks (node) {
...
+      const children = node.children;
+      if (children.length > 1 && children[children.length - 1].tagName === 'br') {
```
Comment (Mr0grog) [inline lib/fix-google-html.js:268]: You can simplify this a bit with optional chaining:

```suggestion
      if (children[children.length - 1]?.tagName === 'br') {
```

Or using `Array.at()` which is in Node 16 (although I’m not totally sure about browser support, but I think it *should* be ok):

```suggestion
      if (children.at(-1)?.tagName === 'br') {
```

File: lib/fix-google-html.js [hunk 3]
```diff
@@ -255,6 +255,28 @@ export function unwrapLineBreaks (node) {
...
+        const endingBr = children.pop()
+        parent.children.splice(index + 1, 0, endingBr);
+      }
+      if (children.length > 0 && children[0].tagName === 'br') {
```
Comment (Mr0grog) [inline lib/fix-google-html.js:272]: Same as above for you this could be simplified:

```suggestion
      if (children[0]?.tagName === 'br') {
```

**Review by Mr0grog — CHANGES_REQUESTED** [review]
Thanks! This looks pretty good — I left a few notes inline, but I think the only *critical* thing here is to clean up that `console.log()` call. 👍

File: lib/fix-google-html.js (hunk 2 above)
Comment (trustedtomato) [reply lib/fix-google-html.js]: I opted for the children.length method, not Array.at is a little bleeding edge, and don't want to risk it until eslint-plugin-compat is merged.

Comment (Mr0grog) [reply lib/fix-google-html.js]: 👍 That sounds reasonable to me.

**Review by Mr0grog — APPROVED** [review]
Thanks for addressing everything so quickly!

**Pull request merged**